Inferences/InductionHelper.cpp

```diff
@@ -53,7 +53,10 @@ bool isIntegerComparisonLiteral(Literal* lit) {
     case Theory::INT_GREATER_EQUAL:
     case Theory::INT_GREATER:
       // All formulas should be normalized to only use INT_LESS and not other integer comparison predicates.
-      ASSERTION_VIOLATION;
+
+      // Equality proxy may generate useless congruence axioms for the likes of INT_GREATER
```

Comment:
I'm not entirely sure they are all useless, but we can certainly do so. If you point me to where it should be done, I can add the flagging. Otherwise, we can leave it to a "cleanup equality proxy symbols" PR.

Reply:
If it's true that we never need the congruence axiom `![W, X, Y, Z]: W = Y & X = Z & W < X => Y < Z`, then I think `sym->markSkipCongruence();` would suffice just above [here](https://github.com/vprover/vampire/blob/master/Kernel/Signature.cpp#L481).

I'm not sure whether this is true! If you're sure, go ahead; if not forget I said anything and we can pick it up later.

Reply:
It feels like this should be checked for all theory symbols by someone qualified at some point. How about coming back to this when cleaning up equality proxy the next time?

Reply:
Seems reasonable to me! Definitely not qualified.